From df934c5149577bc0ef68657644045115bd9d4995 Mon Sep 17 00:00:00 2001
From: Jean Ghali <jghali@libertysurf.fr>
Date: Sun, 19 Sep 2021 11:59:50 +0000
Subject: [PATCH 2/2] #16635: Replace hb_subset, removed in harfbuzz 3.0, by
 hb_subset_or_fail

git-svn-id: svn://scribus.net/trunk/Scribus@24707 11d20701-8431-0410-a711-e3c959e3b870
(cherry picked from commit 68ec41169eaceea4a6e1d6f359762a191c7e61d5)
---
 scribus/fonts/sfnt.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scribus/fonts/sfnt.cpp b/scribus/fonts/sfnt.cpp
index fca8b3915..b3dfa50dc 100644
--- a/scribus/fonts/sfnt.cpp
+++ b/scribus/fonts/sfnt.cpp
@@ -1271,9 +1271,15 @@ namespace sfnt {
 #endif
 #endif
 
+#if HB_VERSION_ATLEAST(2, 9, 0)
+		QScopedPointer<hb_face_t, HbFaceDeleter> hbSubsetFace(hb_subset_or_fail(hbFullFace.get(), hbSubsetInput.get()));
+		if (hbSubsetFace.isNull())
+			return QByteArray();
+#else
 		QScopedPointer<hb_face_t, HbFaceDeleter> hbSubsetFace(hb_subset(hbFullFace.get(), hbSubsetInput.get()));
 		if (hbSubsetFace.isNull())
 			return QByteArray();
+#endif
 
 		QScopedPointer<hb_blob_t, HbBlobDeleter> hbSubsetBlob(hb_face_reference_blob(hbSubsetFace.get()));
 		if (hbSubsetBlob.isNull())
-- 
2.33.0

